pandas/core/frame.py

```diff
@@ -6252,6 +6252,76 @@ def stack(self, level=-1, dropna=True):
         else:
             return stack(self, level, dropna=dropna)
 
+    def explode(self, column: str) -> "DataFrame":
+        """
+        Create new DataFrame expanding a specified list-like column.
+
+        .. versionadded:: 0.25.0
+
+        Parameters
+        ----------
+        column : str
+
+        Returns
+        -------
+        DataFrame
+            Exploded lists to rows of the subset columns;
+            index will be duplicated for these rows.
+
+        Raises
+        ------
+        ValueError :
+            if columns of the frame are not unique.
+
+        See Also
+        --------
+        DataFrame.unstack : Pivot a level of the (necessarily hierarchical)
+            index labels
+        DataFrame.melt : Unpivot a DataFrame from wide format to long format
+        Series.explode : Explode a DataFrame from list-like columns to long format.
+
+        Notes
+        -----
+        This routine will explode list-likes including lists, tuples,
+            Series, and np.ndarray.
+        The result dtype of the subset rows will be object.
+        Scalars will be returned unchanged.
+        Empty list-likes will result in a np.nan for that row.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame({'A': [[1, 2, 3], 'foo', [], [3, 4]], 'B': 1})
+        >>> df
+                   A  B
+        0  [1, 2, 3]  1
+        1        foo  1
+        2         []  1
+        3     [3, 4]  1
+
+        >>> df.explode('A')
+             A  B
+        0    1  1
+        0    2  1
+        0    3  1
+        1  foo  1
+        2  NaN  1
+        3    3  1
+        3    4  1
+
+        """
+
+        if not is_scalar(column):
```

Comment:
Just to make sure I understand implicit assumptions here:

  1. We can't use this with a MultiIndex
  2. We can't use this with custom objects or tuples as labels
  3. We can't use this with duplicate column names because we want to place the exploded column back into the original location 
  4. Column selection is always by label never by index

Not blocking on any of these just want to make sure on the same page

Reply:
1 and 2 should be ok i’ll add some tests

Reply:
Just to clarify I meant a MultiIndex .columns which I don't think this works for. The test you added was for the index - sorry for confusion

Not a big deal for now still good with change as is so @jorisvandenbossche merge if satisfied